### PR TITLE
Implementation for deleting an user from organization

### PR DESCRIPTION
**Pull request by rocknes**
Unit test :
(pyvcloud) C:\code\pyvcloud\tests>python -m unittest vcd_user.TestUser.test_create_and_delete_user
C:\Users\sena\Envs\pyvcloud\lib\site-packages\urllib3-1.22-py3.6.egg\urllib3\connectionpool.py:858: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
None
.
----------------------------------------------------------------------
Ran 1 test in 3.051s

OK

### CODE REVIEW

File: pyvcloud/vcd/org.py [hunk 1]
```diff
@@ -496,6 +472,39 @@ def create_user(self,
...
+        return self.client.get_resource(records[0].get('href'))
+
+    def delete_user(self, user_name):
+        """
+        Delete user record from current organization
+        :param user_name: (str) name of the user that (org/sys)admins wants to delete
+        :return: result of calling DELETE on the user resource
+        """ # NOQA
+        user = self.get_user(user_name)
+        return self.client.delete_resource(user.get('href'))
```
Comment (sahithi) [inline pyvcloud/vcd/org.py]: Where are we deleting user associated entities like vApps, vms, catalogs as described in the vcd-cli command description? Does it force delete automatically?

Comment (rocknes) [reply pyvcloud/vcd/org.py]: vCD does it for us. We don't have to separately delete the entities.

File: pyvcloud/vcd/org.py [hunk 2]
```diff
@@ -537,7 +546,7 @@ def get_roles_query(self, name_filter=None):
         org_filter = None
         if self.is_admin:
             resource_type = 'adminRole'
-            org_filter = 'orgName==%s' % self.resource.get('name')
+            org_filter = 'org==%s' % self.resource.get('href')
```
Comment (anusuyar) [inline pyvcloud/vcd/org.py]: what is the reason to change this?

Comment (rocknes) [reply pyvcloud/vcd/org.py]: Parity with the other areas in code, where we are limiting what the sys admin can see by using href of the org as a filter. The functionality isn't getting impacted, it's mostly cosmetic.

**Pull request merged**